test_conformance/api/test_api_min_max.cpp

```diff
@@ -491,12 +491,12 @@ int test_min_max_mem_alloc_size(cl_device_id deviceID, cl_context context, cl_co
     else
         requiredAllocSize = 128 * 1024 * 1024;
 
-    /* Get the max mem alloc size */
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_MAX_MEM_ALLOC_SIZE, sizeof( maxAllocSize ), &maxAllocSize, NULL );
-    test_error( error, "Unable to get max mem alloc size from device" );
-
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_GLOBAL_MEM_SIZE, sizeof( memSize ), &memSize, NULL );
-    test_error( error, "Unable to get global memory size from device" );
+    /* Get the max mem alloc size, limit the alloc to half of the available
+     * memory */
+    maxAllocSize = get_device_info_max_mem_alloc_size(
+        deviceID, MAX_DEVICE_MEMORY_SIZE_DIVISOR);
+    memSize = get_device_info_global_mem_size(deviceID,
```

Comment:
_requiredAllocSize_ already reduces the memSize by a factor of 4 below. If that value is wrong, it seems like the fix should go there rather than introducing a redundant reduction mechansim.

Reply:
A systematic reduction is done for some tests in allocations : test_conformance/allocations/main.cpp
I agree that we don't want a redundant mechanism but it is better if we have it in a single common helper.

I will review the reduction code already done (the division by a factor of 4) and propose a change to that.

Reply:
Updated the change